src/db/mysql/test.rs

```diff
@@ -0,0 +1,419 @@
+    }
+}
+
+pub fn db() -> MysqlDb {
+    let _ = env_logger::try_init();
+    let settings = Settings {
+        debug: true,
+        port: 8000,
+        database_url: "mysql://megaphone:apple@localhost/syncstorage".to_owned(),
+        database_pool_max_size: Some(1),
+        database_use_test_transactions: true,
+    };
+
+    run_embedded_migrations(&settings).unwrap();
+    let pool = MysqlDbPool::new(&settings).unwrap();
+    pool.get().unwrap()
+}
+
+fn pbso(
+    user_id: u32,
+    cid: i32,
+    bid: &str,
+    payload: Option<&str>,
+    sortindex: Option<i32>,
+    ttl: Option<u32>,
+) -> params::PutBso {
+    params::PutBso {
+        user_id,
+        collection_id: cid,
+        id: bid.to_owned(),
+        payload: payload.map(&str::to_owned),
+        sortindex,
+        ttl,
+        modified: ms_since_epoch(),
+    }
+}
+
+fn gbso(user_id: u32, cid: i32, bid: &str) -> params::GetBso {
+    params::GetBso {
+        user_id,
+        collection_id: cid,
+        id: bid.to_owned(),
+    }
+}
+
+#[test]
+fn static_collection_id() {
+    let db = db();
+
+    // ensure DB actually has predefined common collections
+    let cols: Vec<(i32, _)> = vec![
+        (1, "clients"),
+        (2, "crypto"),
+        (3, "forms"),
+        (4, "history"),
+        (5, "keys"),
+        (6, "meta"),
+        (7, "bookmarks"),
+        (8, "prefs"),
+        (9, "tabs"),
+        (10, "passwords"),
+        (11, "addons"),
+        (12, "addresses"),
+        (13, "creditcards"),
+    ];
+
+    use diesel::{QueryDsl, RunQueryDsl};
+    let results: HashMap<i32, String> = collections::table
+        .select((collections::id, collections::name))
+        .load(&db.conn)
+        .unwrap()
+        .into_iter()
+        .collect();
+    assert_eq!(results.len(), cols.len());
+    for (id, name) in &cols {
+        assert_eq!(results.get(id).unwrap(), name);
+    }
+
+    for (id, name) in &cols {
+        let result = db.get_collection_id_sync(name).unwrap();
+        assert_eq!(result, *id);
+    }
+
+    let cid = db.create_collection_sync("col1").unwrap();
+    assert!(cid >= 100);
+}
+
+#[test]
+fn bso_successfully_updates_single_values() {
+    let db = db();
+
+    let uid = 1;
+    let cid = 1;
+    let bid = "testBSO";
+    let sortindex = 1;
+    let ttl = 3600 * 1000;
+    let bso1 = pbso(
+        uid,
+        cid,
+        bid,
+        Some("initial value"),
+        Some(sortindex),
+        Some(ttl),
+    );
+    db.put_bso_sync(&bso1).unwrap();
+
+    let payload = "Updated payload";
+    let bso2 = pbso(uid, cid, bid, Some(payload), None, None);
+    db.put_bso_sync(&bso2).unwrap();
+
+    let bso = db.get_bso_sync(&gbso(uid, cid, bid)).unwrap().unwrap();
+    assert_eq!(bso.modified, bso2.modified);
+    assert_eq!(bso.payload, payload);
+    assert_eq!(bso.sortindex, Some(sortindex));
+    // XXX: go version assumes ttl was updated here?
+    //assert_eq!(bso.expiry, modified + ttl);
```

Comment:
FWIW I wouldn't expect the ttl to be updated unless you explicitly sent a new TTL in the update.